doc/develop/api/api_conventions.rst

```diff
@@ -0,0 +1,102 @@
+.. _api_conventions.rst:
+
+API Conventions
+###############
+
+Zephyr provides a few programming interfaces that users can use for creating their applications.
+Zephyr's API space includes the following areas:
+- C programming interface (function prototypes, structures, and macros), provided in C header files
+- Configuration system (Kconfig)
+- Device driver model (Devicetree)
+
+Other areas, such as the build system (CMake), may also be seen as Zephyr's API. However, any area
+that is not explicitly listed herein is not considered Zephyr's API as of today.
+
+API classification
+******************
+The files that define the Zephyr APIs may contain symbols intended for different usage. The intended
+API usage is designated by the API class. Zephyr APIs are classified as:
+
+- Private
+    These APIs are intended for use within boundary of a :term:`software component`. Private APIs
+    defined in Zephyr tree are not subject to :ref:`api_lifecycle`. Therefore, can be changed or
+    removed at any time. Changes to the private API may not be documented.
+
+- Internal
+    In general, these APIs are intended for use only between certain software components that are
+    located in Zephyr tree. The context where the API is called or implemented is well defined. For
+    example, functions prefixed with `arch_` are intended for use by the Zephyr kernel to invoke
+    architecture specific code. An API is classified as an Internal on a case by case basis.
+    Internal APIs should not be used out-of-tree code. Internal APIs are not subject to
```

Comment:
may be worth dropping the specific example, I'd imagine a downstream architecture implementation may use/implement some `arch_` calls or something and it's odd to say that they shouldn't use it, there's not alternative. What it's fair to say is that that's the boundary where they should be a ware that they are building on an api that is not considered stable, with all the implications of that.

Reply:
As per process WG 2023-09-13 meeting:

@nashif - officially out of tree architectures are not supported. Internal APIs will not be subject to a lifecycle. There are many examples using internal APIs that need to be fixed.